diff --git a/safe_control_gym/experiments/train_game_controller.py b/safe_control_gym/experiments/train_game_controller.py
index d66c923..da0dd2d 100644
--- a/safe_control_gym/experiments/train_game_controller.py
+++ b/safe_control_gym/experiments/train_game_controller.py
@@ -45,7 +45,7 @@ class Args:
     # Algorithm specific arguments
     env_id: str = "reach_avoid"
     """the id of the environment"""
-    total_timesteps: int = 1e6
+    total_timesteps: int = 1e7
     """total timesteps of the experiments"""
     learning_rate: float = 3e-4
     """the learning rate of the optimizer"""
@@ -152,7 +152,7 @@ if __name__ == "__main__":
     args.minibatch_size = int(args.batch_size // args.num_minibatches)
     args.num_iterations = int(args.total_timesteps // args.batch_size)
     # Hanyang: make saving directory
-    run_name = os.path.join('training_results/' + 'game/ppo/' +f'{args.seed}/' + f'{datetime.now().strftime("%Y.%m.%d_%H:%M")}' )
+    run_name = os.path.join('training_results/' + 'game/ppo/' +f'{args.seed}/' + f'{args.total_timesteps}' )
     if not os.path.exists(run_name):
         os.makedirs(run_name+'/')
         
diff --git a/training_results/game/ppo/2024/2024.06.15_11:12/events.out.tfevents.1718475158.cs-mars-14.29295.0 b/training_results/game/ppo/2024/2024.06.15_11:12/events.out.tfevents.1718475158.cs-mars-14.29295.0
deleted file mode 100644
index 4b365e0..0000000
Binary files a/training_results/game/ppo/2024/2024.06.15_11:12/events.out.tfevents.1718475158.cs-mars-14.29295.0 and /dev/null differ
diff --git a/training_results/game/ppo/2024/2024.06.15_11:12/train_game_controller.cleanrl_model b/training_results/game/ppo/2024/2024.06.15_11:12/train_game_controller.cleanrl_model
deleted file mode 100644
index e5958ff..0000000
Binary files a/training_results/game/ppo/2024/2024.06.15_11:12/train_game_controller.cleanrl_model and /dev/null differ
diff --git a/wandb/latest-run b/wandb/latest-run
index dafbb13..5881952 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20240615_111234-06vgglyi
\ No newline at end of file
+run-20240615_144011-5bzvjoay
\ No newline at end of file
