diff --git a/safe_control_gym/experiments/train_game.py b/safe_control_gym/experiments/train_game.py
index e2174fe..f59782e 100644
--- a/safe_control_gym/experiments/train_game.py
+++ b/safe_control_gym/experiments/train_game.py
@@ -45,7 +45,7 @@ class Args:
     # Algorithm specific arguments
     env_id: str = "reach_avoid"
     """the id of the environment"""
-    total_timesteps: int = 3e7
+    total_timesteps: int = 1e7
     """total timesteps of the experiments"""
     learning_rate: float = 3e-4
     """the learning rate of the optimizer"""
diff --git a/wandb/latest-run b/wandb/latest-run
index fe864e7..50b2c50 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20240619_092047-3oqscu7m
\ No newline at end of file
+run-20240620_125324-8pcg39te
\ No newline at end of file
